Assets/Scripts/Combat/Abilities/SliderHandle.cs

```diff
@@ -0,0 +1,125 @@
+    private BoxCollider2D GoodCollider;
+    private BoxCollider2D PerfectCollider;
+
+    private float SliderSpeed = 0.7f;
+    private int NumberOfClicks;
+    private int MaxClicks = 3;
+    private int GoodClicks;
+    private int PerfectClicks;
+    private bool arrived = false;
+    private bool isDone = false;
+
+    private enum Click { None, Miss, Good, Perfect } 
+
+    private Click click = Click.None;
+
+    private void Start()
+    {
+        //Get Components
+        HitTransform = HitArea.GetComponent<RectTransform>();
+        SliderCollider = slider.GetComponent<BoxCollider2D>();
+        GoodCollider = HitArea.GetComponent<BoxCollider2D>();
+        PerfectCollider = HitArea.transform.GetChild(0).GetComponent<BoxCollider2D>();
+
+        enabled = false;
+    }
+
+    public void StartSlider()
+    {
+        enabled = true;
+        RandomArea(); // Move the hit area randomly
+    }
+
+    private void Update()
+    {
+        if(NumberOfClicks != MaxClicks)
+        {
+            if(!arrived)
+                slider.transform.localPosition = Vector3.MoveTowards(slider.transform.localPosition, end.transform.localPosition, SliderSpeed); // L to R
+            else
+                slider.transform.localPosition = Vector3.MoveTowards(slider.transform.localPosition, start.transform.localPosition, SliderSpeed); // R to L
+
+            if(Input.GetMouseButtonDown(0))
+            {
+                if(SliderCollider.IsTouching(PerfectCollider))
+                {
+                    click = Click.Perfect;
+                    PerfectClicks++;
+                    Debug.Log("Perfect Hit");
+                }
+                else if(SliderCollider.IsTouching(GoodCollider))
+                {
+                    click = Click.Good;
+                    GoodClicks++;
+                    Debug.Log("Good Hit");
+                }
+                else
+                {
+                    click = Click.Miss;
+                    Debug.Log("Missed Hit");
+                }
+
+                NumberOfClicks++;
+                RandomArea();
+            }
+
+            CheckArrived();
+        }
+        else
+        {
+            NumberOfClicks = 0; // Will not call update again because it is stopped in Confection.cs
+            GoodClicks = 0;
+            PerfectClicks = 0;
+        }
+    }
+
+    private void CheckArrived()
+    {
+        if(slider.transform.localPosition == end.transform.localPosition)
+            arrived = true;
+
+        if(slider.transform.localPosition == start.transform.localPosition)
+           arrived = false;
+    }
+
+    private void RandomArea()
+    {
+        Vector2 v = new Vector2(HitTransform.rect.xMin, HitTransform.rect.xMax); // v[0] = min | v[1] = max
+        float randomX = Random.Range(start.transform.localPosition.x + v[1], end.transform.localPosition.x + v[0]);
+        HitArea.transform.localPosition = new Vector3(randomX, slider.transform.localPosition.y, 0.0f);
+    }
```

Comment:
Typically method should be named with verbs; maybe `MoveHitAreaToRandomArea()` or `RandomizeHitAreaPosition()`?

Reply:
Renamed it to RandomizeHitAreaPosition(), Thank you!